contributor_docs/documentation_style_guide.md

```diff
@@ -0,0 +1,1184 @@
+
+for (let i = 0; i < numbers.length; i += 1) {
+  numbersCopy[i] = numbers[i];
+}
+
+// Good.
+const numbersCopy = numbers.slice();
+```
+
+- Write arrays on multiple lines when it improves readibility. Use line breaks after the opening bracket and before the closing bracket. Add a trailing comma.
+
+```javascript
+// Bad.
+const matrix = [[1, 0, 0],
+                [0, 1, 0],
+                [0, 0, 1]];
+
+// Good.
+const matrix = [
+  [1, 0, 0],
+  [0, 1, 0],
+  [0, 0, 1],
+];
+
+// Also good.
+const matrix = [[1, 0, 0], [0, 1, 0], [0, 0, 1]];
+```
+
+**[⬆ back to top](#table-of-contents)**
+
+## Functions
+
+- Use function declarations instead of named function expressions.
+
+> Why? Function declarations have some gotchas, but they are easier to understand for beginners.
+
+```javascript
+// Bad.
+const foo = function () {
+  // ...
+};
+
+// Bad.
+const foo = () => {
+  // ...
+};
+
+// Good.
+function foo() {
+  // ...
+}
+```
+
+- Use default parameter syntax. Don't mutate function arguments.
+
+```javascript
+// Bad.
+function createBall(diameter) {
+  diameter = diameter || 50;
+  // ...
+}
+
+// Good.
+function createBall(diameter = 50) {
+  // ...
+}
+```
+
+- Always put default parameters last.
+
+```javascript
+// Bad.
+function drawSpiral(angle = 90, length) {
+  // ...
+}
+
+// Good.
+function drawSpiral(length, angle = 90) {
+  // ...
+}
+```
+
+**[⬆ back to top](#table-of-contents)**
+
+## Arrow Functions
+
+- Use arrow function notation for anonymous functions. Callbacks are a common case for this syntax.
+
+> Why? The syntax is more concise. It also creates a version of the function that executes in the context of `this`, which is often helpful.
+
+> Why not? If the anonymous function is complex, rewrite it as a declared function.
+
+```javascript
+// Bad.
+loadImage(function (img) {
+  image(img, 0, 0);
+});
+
+// Good.
+loadImage((img) => {
+  image(img, 0, 0);
+});
+
+// Bad.
+loadImage(function (img) {
+  // Complex preprocessing...
+});
+
+// Good.
+loadImage(processImage);
+
+function processImage(img) {
+  // Complex preprocessing...
+}
+```
+
+- Use the implicit return when possible. Omit braces if the function body returns a single statement without side effects. Otherwise, keep the braces and use a `return` statement.
+
+> Why? It's easier to read.
+
+```javascript
+// Bad.
+[1, 2, 3].map((number) => {
+  const squared = number ** 2;
+  `${number} squared is ${squared}.`;
+});
+
+// Bad.
+[1, 2, 3].map((number) => {
+  const squared = number ** 2;
+  return `${number} squared is ${squared}.`;
+});
+
+// Good.
+[1, 2, 3].map((number) => `${number} squared is ${number ** 2}.`);
+```
+
+- Always include parentheses around arguments.
+
+> Why? Doing so reduces bugs when changing parameters.
+
+```javascript
+// Bad.
+[1, 2, 3].map(number => number * number);
+
+// Good.
+[1, 2, 3].map((number) => number * number);
+```
+
+## Classes
+
+- Always use `class`. Avoid manipulating `prototype` directly.
```

Comment:
One possible exception is in the docs for creating libraries, where [we currently recommend adding things to `p5.prototype`](https://github.com/processing/p5.js/blob/main/contributor_docs/creating_libraries.md#you-can-extend-p5-core-functionality-by-adding-methods-to-p5prototype)

Reply:
Good point. I'll note the exception.